### ISSUE TITLE
Optimizer treats variable declaration/assignment using let as syntax error

### ISSUE THREAD

**Issue by wojciech-kaluza-hpe** [requirejs/requirejs#1671]
Attempting to uglify our JS codebase with r.js v. 2.3.3 yielded the following error:

```
Error: Cannot uglify file: C:/dev/git/find/webapp/idol/target/minified-javascript/public.js. Skipping it. Error is:
SyntaxError: Unexpected token: name (config) (line: 20100, col: 8, pos: 637324)
```

The offending line in context is:

```
define('find/app/configuration',[
    'underscore',
    'jquery'
], function(_, $) {
    'use strict';

    let config; // <- line 20100, syntax error

    function parseBooleanOption(config, uiCustomization, option) {
        const optionRules = uiCustomization.options[option];

        return optionRules.user && !(config.hasBiRole && optionRules.bi === false) ||
            optionRules.bi && config.hasBiRole;
    }
//...
}
```

The keyword let causes a failure regardless of context: assignments of type `let varName = 'value';` produce an analogous error.

This error does not occur with the `const` keyword.

**Comment by LeJared** [comment]
I'm having the same issue here. Any news on that?

**Comment by jrburke** [comment]
Issue #959 updates uglifyjs 2.8.29. Uglify 2.8.x does not handle the latest JS syntax though, uglify 3.x does. However, there are no plans to update r.js to use 3.x since its API is different than Uglify 2.x, and I do not want to break people still relying on Uglify 2.x config.

So this is closed as much as what is supported by Uglify 2.8.29.
